airflow-core/src/airflow/ui/src/hooks/usePluginTabs.tsx

```diff
@@ -36,9 +36,11 @@ export const usePluginTabs = (destination: string): Array<TabPlugin> => {
   // Get external views with the specified destination and ensure they have url_route
   const externalViews =
     pluginData?.plugins
-      .flatMap((plugin) => plugin.external_views)
-      .filter((view: ExternalViewResponse) => view.destination === destination && Boolean(view.url_route)) ??
-    [];
+      .flatMap((plugin) => [...plugin.external_views, ...plugin.react_apps])
+      .filter(
+        (view: ExternalViewResponse | ReactAppResponse) =>
```

Comment:
Maybe also a rename to emphasize the fact that this is an API type. 
```suggestion
        (view: NavItemReponse) =>
```